src/Jackalope/Transport/DoctrineDBAL/CachedClient.php

```diff
@@ -476,14 +519,35 @@ public function getWeakReferences($path, $name = null)
     /**
      * {@inheritDoc}
      */
+    public function query(Query $query)
+    {
+        if (empty($this->caches['query'])) {
+            return parent::query($query);
+        }
+
+        $this->assertLoggedIn();
+
+        $cacheKey = "query: {$query->getStatement()}, {$query->getLimit()}, {$query->getOffset()}, {$query->getLanguage()}, ".$this->workspaceName;
```

Comment:
this change is actually part of #280 

to review this maybe better to look at https://github.com/jackalope/jackalope-doctrine-dbal/compare/cache_queries...cache_queries_node_types